dask_planner/src/sql/logical/join.rs

```diff
@@ -20,36 +20,35 @@ pub struct PyJoin {
 impl PyJoin {
     #[pyo3(name = "getCondition")]
     pub fn join_condition(&self) -> PyExpr {
-        // TODO: This logic should be altered once https://github.com/apache/arrow-datafusion/issues/2496 is complete
-        if self.join.on.len() >= 1 {
-            let (left_col, right_col) = &self.join.on[0];
-            let mut root_expr: Expr = Expr::BinaryExpr {
-                left: Box::new(Expr::Column(left_col.clone())),
-                op: Operator::Eq,
-                right: Box::new(Expr::Column(right_col.clone())),
-            };
-            for idx in 1..self.join.on.len() {
-                let (left_col, right_col) = &self.join.on[idx];
-                let ex: Expr = Expr::BinaryExpr {
-                    left: Box::new(Expr::Column(left_col.clone())),
-                    op: Operator::Eq,
-                    right: Box::new(Expr::Column(right_col.clone())),
-                };
+        // equi-join filters
+        let mut filters: Vec<Expr> = self
+            .join
+            .on
+            .iter()
+            .map(|(l, r)| {
+                binary_expr(
+                    Expr::Column(l.clone()),
+                    Operator::Eq,
+                    Expr::Column(r.clone()),
+                )
+            })
+            .collect();
 
-                root_expr = Expr::BinaryExpr {
-                    left: Box::new(root_expr),
-                    op: Operator::Eq,
```

Comment:
This looks like it was incorrect before and should have been `Operator::and`

Reply:
yep, good catch